src/components/Pages/HomeSections/whyUs/WhyUsSection.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import React, { useEffect, createContext, useRef } from "react";
+
+import "./whyUsStyle.scss";
+
+import WhyUsCardContainer from "./WhyUsCardContainer";
+import { LangType } from "./whyUsModel";
+import { useIsVisible } from "@/hooks/useIsVisible";
+
+export const LangContext = createContext<LangType | undefined>(undefined);
+
+export default function WhyUsSection({
+    currentLang,
+    direction,
+}: Readonly<{
+    currentLang: Record<string, string>;
+    direction: "rtl" | "ltr";
+}>) {
+
+    const ref = useRef<HTMLDivElement>(null);
+    const isVisible = useIsVisible(ref);
+
+    useEffect(() => {
+        const body = document.querySelector(".main") as HTMLElement;
+        if (isVisible) {
+            body.classList.add("whyUsActive__main");
+        } else {
+            body.classList.remove("whyUsActive__main");
+        }
+    }, [isVisible]);
+
+
+    return (
+        <LangContext.Provider value={{ currentLang, inSection: isVisible, itemRef:ref }}>
```

Comment:
The object passed as the value prop to the Context provider changes every render. 
Wrap it in a useMemo hook.

Reply:
done